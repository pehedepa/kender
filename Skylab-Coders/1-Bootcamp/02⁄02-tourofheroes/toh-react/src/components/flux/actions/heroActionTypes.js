const heroActionTypes = {
  ADD_HERO: 'ADD_HERO',
  DELETE_HERO: 'DELETE_HERO',
  UPDATE_HERO: 'UPDATE HERO',
  LOAD_HEROES: 'LOAD HEROES',
};

export default heroActionTypes;
