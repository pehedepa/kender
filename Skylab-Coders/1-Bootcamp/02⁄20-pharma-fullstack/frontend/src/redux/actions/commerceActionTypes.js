export default {
  LOAD_ITEMS: 'LOAD_ITEMS',
  ADD_ITEM_TO_CART: 'ADD_ITEM_TO_CART'
};
