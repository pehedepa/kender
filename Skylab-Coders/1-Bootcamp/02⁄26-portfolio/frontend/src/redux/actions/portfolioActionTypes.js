export default {
  GET_GENERIC: 'GET_GENERIC',
  GET_ACADEMIC: 'GET_ACADEMIC',
  GET_TRADESKILLS: 'GET_TRADESKILLS',
  GET_WORKEXPERIENCE: 'GET_WORKEXPERIENCE'
};
