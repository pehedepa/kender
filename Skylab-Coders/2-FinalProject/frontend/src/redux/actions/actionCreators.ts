import axios from 'axios'
import actionTypes from './actionTypes'
import { loginURL, registerURL, productsURL, usersURL, logoutURL } from '../../constants'
import { Dispatch } from 'react'

export function login (email: string, password: string): Promise<any> {
  const data = { email, password }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.post(loginURL, data)
      dispatch({
        type: actionTypes.LOGIN_USER,
        payload: response
      })
    } catch (error) {
      dispatch({
        type: actionTypes.LOGIN_ERROR,
        payload: 401
      })
    }
  }
}

export function createAcc (email: string, password: string, location: string, name: string): Promise<any> {
  const data = { email, password, location, name }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.post(registerURL, data)
      dispatch({
        type: actionTypes.ADD_USER,
        payload: response.status
      })
    } catch (error) {
      dispatch({
        type: actionTypes.ADD_USER_ERROR,
        payload: 401
      })
    }
  }
}

export function uploadProduct (name: string, price: number, description: string, image: string, user: string, category: string, ageRange: string, keywords: any[]): Promise<any> {
  const data = { name, price, description, image, user, category, ageRange, keywords }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.post(productsURL, data)
      dispatch({
        type: actionTypes.UPLOAD_PRODUCT,
        payload: response.status
      })
    } catch (error) {
      dispatch({
        type: actionTypes.UPLOAD_PRODUCT,
        payload: 401
      })
    }
  }
}

export function getAllProducts (): Promise<any> {
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.get(productsURL)
      dispatch({
        type: actionTypes.GET_ALL_PRODUCTS,
        payload: response.data
      })
    } catch (error) {
      dispatch({
        type: actionTypes.GET_ALL_PRODUCTS_ERROR,
        payload: 401
      })
    }
  }
}

export function uploadProfile (_id, photo, name, address, cc, location): Promise<any> {
  const data = { _id, photo, name, address, cc, location }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.put(usersURL, data)
      dispatch({
        type: actionTypes.UPDATE_PROFILE,
        payload: response
      })
    } catch (error) {
      dispatch({
        type: actionTypes.UPDATE_PROFILE,
        payload: 401
      })
    }
  }
}

export function logout (email, password): Promise<any> {
  const data = { email, password }
  return async function dispatchInfo (dispatch) {
    try {
      await axios.post(logoutURL, data)
      dispatch({
        type: actionTypes.LOG_OUT,
        payload: null
      })
    } catch (error) {
      dispatch({
        type: actionTypes.LOG_OUT,
        payload: null
      })
    }
  }
}

export function deleteProduct (_id) {
  return async function dispatchInfo (dispatch) {
    try {
      await axios.delete(`${productsURL}/${_id}`)
      dispatch({
        type: actionTypes.DELETE_PRODUCT
      })
    } catch (error) {
      dispatch({
        type: actionTypes.DELETE_PRODUCT,
        payload: null
      })
    }
  }
}

export function applyFilter (data: string) : { type: string, payload: string } {
  return {
    type: actionTypes.FILTER_PRODUCTS,
    payload: data
  }
}

export function cleanFilters (): { type: string, payload: null } {
  return {
    type: actionTypes.CLEAN_FILTERS,
    payload: null
  }
}

export function cleanRegister (): { type: string } {
  return {
    type: actionTypes.CLEAN_REGISTER_STATUS
  }
}

export function cleanLogin (): { type: string } {
  return {
    type: actionTypes.CLEAN_LOGIN_STATUS
  }
}

export function cleanProductUploaded (): { type: string } {
  return {
    type: actionTypes.CLEAN_UPLOAD_PRODUCT
  }
}
