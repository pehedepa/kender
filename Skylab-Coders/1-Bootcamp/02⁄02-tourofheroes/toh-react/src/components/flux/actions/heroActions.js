import dispatcher from '../dispatcher/dispatcher';
import heroActionTypes from './heroActionTypes';

export function saveHero(hero) {
  debugger;
  const action = {
    type: hero.id ? heroActionTypes.UPDATE_HERO : heroActionTypes.ADD_HERO,
    hero,
  };

  dispatcher.dispatch(action);
}

export function deleteHero({ id }) {
  debugger;
  const action = {
    type: heroActionTypes.DELETE_HERO,
    heroId: id,
  };

  dispatcher.dispatch(action);
}

export function loadHero(heroId) {
  const action = {
    type: heroActionTypes.LOAD_HERO,
    heroId,
  };

  dispatcher.dispatch(action);
}
