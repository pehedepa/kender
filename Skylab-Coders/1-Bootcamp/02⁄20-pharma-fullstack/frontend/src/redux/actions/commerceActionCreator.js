import axios from 'axios';
import commerceActionTypes from './commerceActionTypes';

const api = 'http://localhost:5000/api/commerce';

export default function getItems() {
  return async (dispatch) => {
    const { data } = await axios(api);
    dispatch({
      type: commerceActionTypes.LOAD_ITEMS,
      data
    });
  };
}

export function pushItemToCart(name, _id, url, stock, price) {
  const payload = {
    name,
    _id,
    url,
    stock,
    price
  };

  return (dispatch) => {
    dispatch({
      type: commerceActionTypes.ADD_ITEM_TO_CART,
      data: payload
    });
  };
}
