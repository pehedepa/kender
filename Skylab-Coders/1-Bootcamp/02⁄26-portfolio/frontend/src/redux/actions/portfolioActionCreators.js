import axios from 'axios';
import portfolioActionTypes from './portfolioActionTypes';

const api = 'http://localhost:5000/api/portfolio';

export default function getGenericData() {
  const generic = '/generic';
  return async (dispatch) => {
    const { data } = await axios(api + generic);
    dispatch({
      type: portfolioActionTypes.GET_GENERIC,
      data
    });
  };
}

export function getTradeSkillsData() {
  const tradeSkills = '/tradeskills';
  return async (dispatch) => {
    const { data } = await axios(api + tradeSkills);
    dispatch({
      type: portfolioActionTypes.GET_TRADESKILLS,
      data
    });
  };
}
