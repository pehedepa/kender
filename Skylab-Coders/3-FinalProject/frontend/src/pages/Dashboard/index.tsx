import React, { useEffect } from 'react'
import { View, Text } from 'react-native'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getAllProducts } from '../../redux/actions/actionCreators'
import Header from '../../components/Header'
import Products from '../../components/Products'
import Categories from '../../components/Categories'
import Loading from '../../components/Loading'
import styles from './styles'
import { AppStyles } from '../../styles/theme.styles'

function Dashboard ({ navigation, productList, filteredProducts, actions }: any): any {
  useEffect(() => {
    actions.getAllProducts()
  }, [filteredProducts])

  return (
    <>
        <Header navigation={navigation} />
        <Categories />
        <Text style={styles.title}>LATEST PRODUCTS UPLOADED</Text>
        <View style={[AppStyles.container, styles.container, AppStyles.bgGray]}>
        {Boolean(!productList.length) && (
          <Loading />
        )}
          <Products listItems={(productList.length && !filteredProducts.length) ? productList : filteredProducts} navigation={navigation}/>
        </View>
      </>
  )
}

function mapStateToProps (state): { productList: any[], filteredProducts: any[] } {
  return {
    productList: state.productList,
    filteredProducts: state.filteredProducts
  }
}

function mapDispatchToProps (dispatch): { actions: any } {
  return {
    actions: bindActionCreators({ getAllProducts }, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard)
