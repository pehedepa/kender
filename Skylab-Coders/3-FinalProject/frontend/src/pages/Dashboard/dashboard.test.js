import React from 'react'
import Dashboard from './index'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import * as actions from '../../redux/actions/actionCreators'
import { render } from '@testing-library/react-native'
import thunk from 'redux-thunk'

jest.mock('@react-navigation/native')
jest.mock('../../components/Header', () => 'HeaderMocked')
jest.mock('../../components/Categories', () => 'CategoriesMocked')
jest.mock('../../components/Loading', () => 'LoadingMocked')

describe('Given a Dashboard function', () => {
  const mockStore = configureStore([thunk])
  let store
  let navigation
  beforeEach(() => {
    jest.spyOn(actions, 'getAllProducts').mockReturnValueOnce({ type: '' })
    store = mockStore({
      productList: {},
      filteredProducts: {}
    })
    navigation = {
      navigate: jest.fn()
    }
  })

  describe('When we invoke Dashboard function', () => {
    test('Then the render should match its snaptshot', () => {
      const tree = render(<Provider store={store}><Dashboard navigation={navigation}></Dashboard></Provider>)

      expect(tree).toMatchSnapshot()
    })
  })
})
