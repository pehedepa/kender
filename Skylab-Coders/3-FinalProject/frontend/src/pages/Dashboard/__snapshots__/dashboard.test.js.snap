// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Dashboard function When we invoke Dashboard function Then the render should match its snaptshot 1`] = `
Array [
  <HeaderMocked
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />,
  <CategoriesMocked />,
  <Text
    style={
      Object {
        "color": "#2F2C23",
        "fontFamily": "MontserratMedium",
        "fontSize": 14,
        "marginVertical": 10,
        "paddingLeft": 14,
      }
    }
  >
    LATEST PRODUCTS UPLOADED
  </Text>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "padding": 25,
          "width": "100%",
        },
        Object {
          "padding": 10,
        },
        Object {
          "backgroundColor": "#f7f7f7",
        },
      ]
    }
  >
    <LoadingMocked />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    />
  </View>,
]
`;
